client/extensions/woocommerce/state/wc-api/product-categories/actions.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Internal dependencies
+ */
+import wp from 'lib/wp';
+import { error } from '../actions';
+import {
+	WOOCOMMERCE_API_PRODUCT_CATEGORIES_GET,
+	WOOCOMMERCE_API_PRODUCT_CATEGORIES_GET_SUCCESS,
+} from '../../action-types';
+
+export function getProductCategories( siteId ) {
+	return ( dispatch ) => {
+		const getAction = {
+			type: WOOCOMMERCE_API_PRODUCT_CATEGORIES_GET,
+			payload: { siteId },
+		};
+
+		dispatch( getAction );
+
+		const jpPath = `/jetpack-blogs/${ siteId }/rest-api/`;
+		const apiPath = '/wc/v2/products/categories';
+
+		return wp.req.get( { path: jpPath }, { path: apiPath } )
```

Comment:
Yeah, I will definitely make some helper functions for that. But I wanted a couple more examples of what is needed first.